Chapters 1 & 2 Revisions 05-02-2025 (#5113)
dfinity/motoko: doc/md/fundamentals/1-basic-syntax/2-imports.md

# Imports

> In Motoko, related code modules are organized into packages. Modules can be imported either from named packages or from the local file system using relative paths. The compiler locates packages on the file system based on a command-line argument specifying their location.

Reviewer: Command line is not hyphenated. https://github.com/dfinity/portal/blob/master/style-guide.md#language

Imports should be placed at the top of the source file. They enable code reuse from external libraries or modules, helping to improve maintainability and organization. You can import from:

**1. Standard modules provided by the base library.**

```motoko no-repl
import Text "mo:base/Text";
import Nat "mo:base/Nat";
import Math "mo:base/Float";
```

The package `base` is Motoko's standard library.
This imports the `Text`, `Nat` and `Float` modules from package `base` under the local names `Text`, `Nat` and `Math`.

> It's good practice, but not required, to import a module using its name in the package.

Reviewer: ```suggestion
While not required, it's considered good practice to import a module using its package-defined name. This helps with consistency and readability across codebases.
```

**2. Packages installed via a package manager (such as Mops).**

```motoko no-repl

> import Nat "mo:base/Nat";

Reviewer 2: ```suggestion
import Iter "mo:base/itertools";
```

```

> This imports the `Nat` and `Float` modules from package `base` under locals names `Nat` and `Math`

Reviewer 2: ```suggestion
```

The module `Iter` is imported from a third-party package `itertools`.

> **3. Files within the current project.**

Reviewer 2: ```suggestion

**3. Files within the current project.**
```

import Utils "Utils";
```

> **You can also import specific functions from a module:**

Author: This is just saying you can import both an actor reference and an actor class instance. Might be better to remove it for now
